mersenne-31/src/lib.rs

```diff
@@ -222,6 +224,7 @@ impl Neg for Mersenne31 {
 impl Mul for Mersenne31 {
     type Output = Self;
 
+    #[allow(clippy::cast_possible_truncation)]
```

Comment:
Alternatively, you could use the bytemuck crate to cast `u64` to `[i32; 2]`.

There's also `widening_mul`, but it's unstable.

Reply:
I think the `allow` as you have it sounds good here 👍 With the current impl it seems like we don't quite want to split into two `u32`s, since `prod_hi` includes one bit from the low `u32`. I think a prior impl version did split into two `u32`s; it might have been a bit slower (even though the split is free) but I didn't actually time them.

Reply:
Oh, yes, I noticed that we are only right shifting by 31.